### ISSUE TITLE
ValueError: invalid literal for int() with base 10: b''

### ISSUE THREAD

**Issue by djdookie** [hiddenswitch/Spellsource-Server#740]
Platform: Windows 10, Anaconda Python 3.6 environment

Reproduction steps:

 1. Folowed the manual "Quick Start Python"
 2. Ran https://github.com/hiddenswitch/Spellsource-Server/blob/master/docs/simulation_example.ipynb
 3. Got the following error message after step `ctx = Context()`:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-a940b997b5b1> in <module>
----> 1 ctx = Context()

~\AppData\Local\conda\conda\envs\python36\lib\site-packages\spellsource\context.py in __init__(self)
     47         except Exception as ex:
     48             self.status = Context.STATUS_FAILED
---> 49             raise ex
     50 
     51         for name, package in (('game', 'net.demilich.metastone.game.*'),

~\AppData\Local\conda\conda\envs\python36\lib\site-packages\spellsource\context.py in __init__(self)
     41         self._is_closed = False
     42         try:
---> 43             self._gateway = Context._start_gateway()
     44         except FileNotFoundError:
     45             self.status = Context.STATUS_FAILED

~\AppData\Local\conda\conda\envs\python36\lib\site-packages\spellsource\context.py in _start_gateway()
    169                                         "--add-opens", "java.management/sun.management=ALL-UNNAMED",
    170                                         "--add-opens", "jdk.management/com.sun.management.internal=ALL-UNNAMED"],
--> 171                               die_on_exit=True)
    172 
    173         # connect python side to Java side with Java dynamic port and start python

~\AppData\Local\conda\conda\envs\python36\lib\site-packages\py4j\java_gateway.py in launch_gateway(port, jarpath, classpath, javaopts, die_on_exit, redirect_stdout, redirect_stderr, daemonize_redirect, java_path, create_new_process_group, enable_auth)
    325     # Determine which port the server started on (needed to support
    326     # ephemeral ports)
--> 327     _port = int(proc.stdout.readline())
    328 
    329     # Read the auth token from the server if enabled.

ValueError: invalid literal for int() with base 10: b''
```

**Comment by djdookie** [comment]
I see some similarities with https://github.com/baztian/jaydebeapi/issues/32 (same command, same error) but I can't figure out whats the problem and how to solve it.

**Comment by djdookie** [comment]
Man, I solved the issue myself after diving into py4j pretty deep and for many hours.
I'd like to share my solution in detail with anyone else, because the reason was pretty weired and unobvious and I could possibly save you the time and pain if you have the same problem.

In short: It was my environment. It was related to a 4 year old earlier installation of java.

**Solution: Remove java.exe, javaw.exe and javaws.exe from Windows\System32 and Windows\SysWOW64!**

Longer story:

I analysed the code in py4j which is opening the java-gateway (that's def launch_gateway(...) in java_gateway.py), especially when the java subprocess is being started with
`proc = Popen(command, stdout=PIPE, stdin=PIPE, stderr=stderr, **popen_kwargs)`

The command sequence looked like (I was using a fixed port 50005 for testing purposes):
`'java -classpath C:\\ProgramData\\Anaconda3\\share/py4j/py4j0.10.8.1.jar;C:\\ProgramData\\Anaconda3\\share\\spellsource\\net-0.7.8-all.jar --add-modules java.se --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens java.management/sun.management=ALL-UNNAMED --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED py4j.GatewayServer --die-on-broken-pipe 50005'`

After reducing the command and trying a simple
```
cmd = 'java'
with Popen([cmd], stdout=PIPE, text=True) as proc:
    print(proc.stdout.read())
```
I got the following error:
```
Error: opening registry key 'Software\JavaSoft\Java Runtime Environment'
Error: could not find java.dll
Error: Could not find Java SE Runtime Environment.
```
Now I finally had a clue what was wrong, and I quick google search finally led me to a stackoverflow thread where a similar problem and the solution was described:

https://stackoverflow.com/questions/6362037/java-error-opening-registry-key/7874543

What a hunt...

@doctorpangloss Thanks for your support on this. I am so glad that I can finally use your great work now. ;)
